### DISCUSSION TITLE
mini.align question

### DISCUSSION THREAD

**Discussion by albertfgu** [nvim-mini/mini.nvim#329] [Q&A] [answered]
I recently migrated from vim-easy-align to mini.align; thanks for the amazing plugins! It's working out of the box for my use case, except for one minor detail that's confusing me. When I align a table on `&`, all the columns are shrunk to the smallest possible width, except the first column is much larger than expected. See screenshot:

![image](https://github.com/echasnovski/mini.nvim/assets/8164679/5b57e82b-1032-4d99-aeea-abf095d5775d)

I apologize if this question has been answered in other documentation

**Comment by echasnovski** [comment, marked as answer]
Columns are not shrunk, but the opposite - they are padded (from left and/or right). Try using `trim` step with custom merge character. 

That is:
- Visually select region to be aligned.
- Press `gA` to initiate interactive align.
- Press `&` to align by it.
- Press `t` to trim whitespace.
- Press `m` followed by `<Space>` and `<CR>` (Enter).

**Comment by albertfgu** [comment]
Thanks for the fast response. This seems to add one extra whitespace to that first column which is fine. Next question is: is there a way to create "recipes" by binding all of that to one key after `ga` (e.g. override the behavior of `ga&`)?

**Reply by echasnovski** [reply]
> Thanks for the fast response. This seems to add one extra whitespace to that first column which is fine. Next question is: is there a way to create "recipes" by binding all of that to one key after `ga` (e.g. override the behavior of `ga&`)?

Yes, there is. Through writing own modifier. That is, adding a function to `config.modifiers` following [this help entry](https://github.com/echasnovski/mini.nvim/blob/main/doc/mini-align.txt#L480). My previous suggestion translates like so:

```lua
require('mini.align').setup({
  modifiers = {
    ['&'] = function(steps, opts)
      opts.split_pattern = '&'
      table.insert(steps.pre_justify, MiniAlign.gen_step.trim())
      opts.merge_delimiter = ' '
    end
  }
})
```
